### PR TITLE
feat(medusa): allow creating draft orders without requiring an email address

### LINKED ISSUE #2686: Draft order email should be optional
**Is your feature request related to a problem? Please describe.**
I want to make an anonymous sale, in-person, where the customer may not want to provide an email.

**Describe the solution you'd like**
Draft order API shouldn't require an email

### PR DESCRIPTION
**Pull request by josetr**
Please let me know if I need to make changes in more places altho I can't build the project due to #2703

Fixes #2686

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/draft-orders/create-draft-order.ts [hunk 1]
```diff
@@ -242,6 +241,7 @@ export class AdminPostDraftOrdersReq {
   status?: string
 
   @IsEmail()
+  @IsOptional()
   email: string
```
Comment (adrien2p) [inline packages/medusa/src/api/routes/admin/draft-orders/create-draft-order.ts:245]: suggestion: Missing `?`

File: packages/medusa/src/api/routes/admin/draft-orders/create-draft-order.ts [hunk 2]
```diff
@@ -34,7 +34,6 @@ import { IsType } from "../../../../utils/validators/is-type"
  *     application/json:
```
Comment (adrien2p) [inline packages/medusa/src/api/routes/admin/draft-orders/create-draft-order.ts:34]: Question: would it required some other changes around the service layer as well?

**Review by adrien2p — CHANGES_REQUESTED** [review]

**Comment by langovoi** [comment]
Any updates here?

**Comment by josetr** [comment]
@olivermrbl @adrien2p can you guys take a look at this one and #2714 again?

**Comment by josetr** [comment]
🙏

**Comment by riqwan** [comment]
Hey, thanks for the PR! Since v2 brought a lot of architectural and API changes on the backend, we will be closing this ticket.

**Pull request closed without merging**